### PR TITLE
Update for multi-container environment

### PR DESCRIPTION
**Pull request by Omicron7**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,23 +1,31 @@
...
+
+# Install rails/app dependencies
+RUN apk --no-cache add libc6-compat git postgresql-libs tzdata nodejs yarn
+
+WORKDIR /usr/src/app
 
 COPY Gemfile Gemfile.lock ./
 
+# Install bundler version which created the lock file
+RUN gem install bundler -v $(awk '/^BUNDLED WITH/ { getline; print $1; exit }' Gemfile.lock)
```
Comment (jbirdjavi) [inline Dockerfile:19]: This isn't strictly necessary anymore. The latest versions of bundler (>2.3) will automatically find and download the correct version of bundler for app, based on Gemfile.lock.
https://bundler.io/blog/2022/01/23/bundler-v2-3.html

Plus, if Gemfile.lock changes, then will the build have to invalidate the cache back to this line?

Comment (jbirdjavi) [reply Dockerfile:19]: Oh I see it would do that anyway at line 16.

File: Dockerfile [hunk 2]
```diff
@@ -34,10 +42,12 @@ ARG DB_ENV_POSTGRESQL_PASS
 ARG DB_PORT_5432_TCP_ADDR
 ARG SITE_URL
 
-RUN bundle exec rake assets:clobber assets:precompile RAILS_ENV=production
+# Compile assets
+RUN RAILS_ENV=production bundle exec rake assets:clobber assets:precompile
 
-# Run this last to make sure permissions are all correct
-RUN mkdir -p /home/app/webapp/tmp /home/app/webapp/db /home/app/webapp/log /home/app/webapp/public/uploads && \
-  chmod -R ugo+rw /home/app/webapp/tmp /home/app/webapp/db /home/app/webapp/log /home/app/webapp/public/uploads
```
Comment (jbirdjavi) [inline Dockerfile:41]: Is this not necessary anymore because we're running rails as root now?

File: Dockerfile (hunk 1 above)
Comment (Omicron7) [reply Dockerfile:19]: If Gemfile.lock changes, cache is invalidated back to the previous line already where it copies the lock file to the container.

File: Dockerfile (hunk 2 above)
Comment (Omicron7) [reply Dockerfile:41]: We aren't using `/home/app/webapp` at all now, app is now in `/usr/src/app`. As far as permissions, I'm not sure. We'll need to test some uploads and see. rails-canary worked fine without this.

File: Gemfile.lock [hunk 3]
```diff
@@ -249,28 +257,29 @@ GEM
...
-      multi_json (~> 1.3)
       multi_xml (~> 0.5)
-      rack (>= 1.2, < 3)
+      rack (>= 1.2, < 4)
+      snaky_hash (~> 2.0)
+      version_gem (~> 1.1)
     oj (3.7.12)
     oj_mimic_json (1.0.1)
-    omniauth (1.9.2)
+    omniauth (2.1.0)
```
Comment (jbirdjavi) [inline Gemfile.lock:276]: Omniauth gems were upgraded?

**Review by jbirdjavi — APPROVED** [review]

**Review by jbirdjavi — APPROVED** [review]

**Pull request merged**